src/HttpServer.gren

```diff
@@ -0,0 +1,332 @@
+effect module HttpServer where { command = HttpCmd, subscription = HttpSub } exposing
+    ( ServerError(..)
+    , Request
+    , Server
+    , Method(..)
+    , createServer
+    , initialize
+    , onRequest
+    , toRequest
+    , requestInfo
+    , respond
+    )
+
+
+import Bytes exposing (Bytes)
+import Bytes.Decode as Decode
+import Dict exposing (Dict)
+import Init
+import Internal.Init
+import Node
+import Task exposing (Task)
+import Gren.Kernel.HttpServer
+import HttpServer.Response exposing (Response(..))
+import Url exposing (Url, Protocol(..))
+
+
+-- TODO: docs!
+-- Preview with:
+--   npm install -g gren-packages
+--   gren-doc-preview
+
+
+type Server
+    -- Note: Actual implementation in Kernel code
+    = Server
+
+
+type alias Request =
+    { headers : Dict String String
+    , method : Method
+    , body : Bytes
+    , bodyAsString : Maybe String
```

Comment:
I think it would be better to have a function called `bodyAsString` that takes in a `Request`.

That way you only pay the decoding cost if the result is actually needed.